Mathlib/NumberTheory/Divisors.lean

```diff
@@ -58,6 +58,15 @@ def divisorsAntidiagonal : Finset (ℕ × ℕ) :=
   (Icc 1 n).filterMap (fun x ↦ let y := n / x; if x * y = n then some (x, y) else none)
     fun x₁ x₂ (x, y) hx₁ hx₂ ↦ by aesop
 
+/-- Pairs of divisors of a natural number as a list.
+
+`n.divisorsAntidiagonalList` is the list of pairs `(a, b) : ℕ × ℕ` such that `a * b = n`, ordered
+by increasing `a`. By convention, we set `Nat.divisorsAntidiagonalList 0 = []`.
```

Comment:
How much work is it to prove that the list is sorted under `Prod.fst`?

Reply:
Not too bad actually, once one has proven the following (I've added a proof in this PR, but the lemma will of course need to be moved to another file).
```
lemma List.Sorted.filterMap {α β : Type*} [DecidableEq α] [DecidableEq β]
    (p : α → Option β) (l : List α)
    (r : α → α → Prop) (r' : β → β → Prop) (hl : l.Sorted r)
    (hp : ∀ (a b : α) (c d : β), p a = some c → p b = some d → r a b → r' c d) :
    (l.filterMap p).Sorted r'
```
One can easily use this to prove that `divisorsAntidiagonalList` is ordered under `Prod.Lex`.